frontend/app/pages/dashboards/dashboard-list.js

```diff
@@ -0,0 +1,100 @@
+import template from './dashboard-list.html';
+import {_} from 'underscore';
+
+function DashboardListCtrl($scope, Dashboard, $location, currentUser, clientConfig, NgTableParams) {
+  const self = this;
+
+  this.logoUrl = clientConfig.logoUrl;
+  const page = parseInt($location.search().page || 1, 10);
+  const count = 25;
+
+  this.defaultOptions = {};
+  this.dashboards = Dashboard.query({}); // shared promise
+
+  $scope.selectedTags = []; // in scope because it needs to be accessed inside a table refresh
+  $scope.searchText = "";
+
+  $scope.$watch(function(){
+    return $scope.searchText;
+  }, function(){
+    if(this){
+      this.defaultOptions.reload()
+    }
+  });
+
+  this.tagIsSelected = (tag) => {
+    return $scope.selectedTags.indexOf(tag) > -1;
+  }
+
+  this.toggleTag = (tag) => {
+    if(this.tagIsSelected(tag)){
+      $scope.selectedTags = $scope.selectedTags.filter((e) => e!=tag);
+    }else{
+      $scope.selectedTags.push(tag);
+    }
+    this.tableParams.reload();
+  }
+
+  this.allTags = [];
+  self.dashboards.$promise.then((data) => {
+    const out = data.map((dashboard) => {
+      return dashboard.name.match(/(^\w+):|(#\w+)/ig);
+    });
+    this.allTags = _.unique(_.flatten(out)).filter((e) => e);
+  });
+
+  this.tableParams = new NgTableParams({ page, count }, {
+    getData(params) {
+      const options = params.url();
+      $location.search('page', options.page);
+
+      const request = {};
+
+      return self.dashboards.$promise.then((data) => {
```

Comment:
If I understand this right, it means we will load the same dashboards from the server over and over again, whenever the user changes tags selection/search term. 

Let's load the dashboards once, and then only re-apply the filters.

Maybe use the router's resolve mechanism here to make sure we have dashboards loaded before the controller gets called?